src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -1495,6 +1495,45 @@
       (sql/do-commands
        "ALTER TABLE certnames ADD CONSTRAINT certnames_reports_id_fkey FOREIGN KEY (latest_report_id) REFERENCES reports(id) ON DELETE SET NULL")))
 
+
+(defn change-value-integer-to-numeric
+  []
+  (let [hash-type (if (sutils/postgres?) "bytea" "varchar(40)")
+        munge-hash (if (sutils/postgres?)
+                     (fn [column] (format "('\\x' || %s)::bytea" column)) identity)]
+    (sql/create-table :fact_values_transform
+                      [ "id" "bigint NOT NULL DEFAULT nextval('fact_values_id_seq')"]
+                      ["value_hash" hash-type "NOT NULL"]
+                      ["value_type_id" "bigint NOT NULL"]
+                      ["value_integer numeric(1000,0)"]
```

Comment:
Looks like the maximum definable "up front" is 1000, otherwise it's 131072 digits. Is that why you chose 1000?

I think that's about 3320 bits of storage, about 415 bytes (without storage overhead). The maximum that decimal can take for a non-float is 435410 bits, about 55 kb per cell.